### PR TITLE
cli: add command for script project initialization

### PR DESCRIPTION
**Pull request by FeelyChau**

### CODE REVIEW

**Review by yorkie — APPROVED** [review]

File: packages/cli/src/bin/pipcook.ts [hunk 1]
```diff
@@ -3,12 +3,19 @@
...
+import * as fs from 'fs-extra';
 import { join } from 'path';
 import { constants } from '@pipcook/core';
 import { readJson, mkdirp, remove } from 'fs-extra';
 import { StandaloneRuntime } from '../runtime';
-import { logger, dateToString } from '../utils';
+import { logger, dateToString, execAsync } from '../utils';
+const download = require('download-git-repo');
 
+const templateMap: Record<string, any> = {
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:14]:
```suggestion
const templateMap: Record<string, string> = {
```

File: packages/cli/src/bin/pipcook.ts [hunk 2]
```diff
@@ -3,12 +3,19 @@
...
 import * as chalk from 'chalk';
 import * as program from 'commander';
+import * as fs from 'fs-extra';
 import { join } from 'path';
 import { constants } from '@pipcook/core';
 import { readJson, mkdirp, remove } from 'fs-extra';
 import { StandaloneRuntime } from '../runtime';
-import { logger, dateToString } from '../utils';
+import { logger, dateToString, execAsync } from '../utils';
+const download = require('download-git-repo');
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:12]:
```suggestion
const download = import('download-git-repo');
```

File: packages/cli/src/bin/pipcook.ts [hunk 3]
```diff
@@ -45,6 +52,39 @@ export const cacheClean = async (opts: CacheCleanOptions): Promise<void> => {
...
+  try {
+    const destDir = join(process.cwd(), name);
+    const pkgFile = join(destDir, 'package.json');
+    if (await fs.pathExists(destDir)) {
+      throw new TypeError(`${name} already exists`);
+    }
+    const [ type, branch = 'main' ] = scriptType.split('@');
+    const template = `${templateMap[type]}#${branch}`;
+    if (!template) {
+      throw new TypeError(`no template found for ${scriptType}, it should be one of 'datasource', 'dataflow', 'model'`);
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:65]: It's good to generate the string by `templateMap`.

File: packages/cli/src/bin/pipcook.ts [hunk 4]
```diff
@@ -45,6 +52,39 @@ export const cacheClean = async (opts: CacheCleanOptions): Promise<void> => {
...
+    }
+    await new Promise<void>((resolve, reject) => {
+      download(template, destDir, { clone: true }, function (err: Error) {
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:68]: Why not using `utils.promisify`?

File: packages/cli/src/bin/pipcook.ts [hunk 5]
```diff
@@ -45,6 +52,39 @@ export const cacheClean = async (opts: CacheCleanOptions): Promise<void> => {
   await Promise.all(futures);
 };
 
+export const createScriptRepo = async (scriptType: string, name: string): Promise<void> => {
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:55]: This function could be finished by `npm init`, see https://docs.npmjs.com/cli/v7/commands/npm-init/.

**Review by yorkie — CHANGES_REQUESTED** [review]

File: packages/cli/src/bin/pipcook.ts (hunk 2 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook.ts]: `download-git-repo` has not types, so if `import` it, we will get error `ts(7016)`.

File: packages/cli/src/bin/pipcook.ts (hunk 5 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook.ts:54]: Does that mean we need to maintain another project named `create-pipcook-script`?

**Comment by FeelyChau** [comment]
Add more details in the help description of command `pipcook script`, @yorkie could you review again, plz?

File: packages/cli/src/bin/pipcook.ts (hunk 5 above)
Comment (yorkie) [reply packages/cli/src/bin/pipcook.ts:61]: What we need is to rename our template to create-* projects.

File: packages/cli/src/bin/pipcook.ts (hunk 3 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook.ts]: Can you describe it more?

File: packages/cli/src/bin/pipcook.ts (hunk 2 above)
Comment (yorkie) [reply packages/cli/src/bin/pipcook.ts]: We need to leave a comment for this.

File: packages/cli/src/bin/pipcook.ts [hunk 6]
```diff
@@ -3,11 +3,24 @@
...
-import { logger, dateToString } from '../utils';
+import { logger, dateToString, execAsync, downloadFromGit } from '../utils';
+
+const templateMap: Record<string, string> = {
+  'datasource': 'imgcook/pipcook-datasource-template',
+  'dataflow': 'imgcook/pipcook-dataflow-template',
+  'model': 'imgcook/pipcook-model-template'
+};
+
+const scriptTypeDescription = 'scriptType should be one of \'datasource<@branch>\', \'dataflow<@branch>\', \'model<@branch>\'.\n\
```
Comment (yorkie) [inline packages/cli/src/bin/pipcook.ts:19]: Why not generating the string from `templateMap`?

File: packages/cli/src/bin/pipcook.ts (hunk 5 above)
Comment (FeelyChau) [reply packages/cli/src/bin/pipcook.ts:61]: If that, the create-* projects must be node packages first? Does it support branch?

**Comment by FeelyChau** [comment]
We have a Pipcook script initializer now:
```sh
$ npm init pipcook-script datasource my-script -t ts
```
This PR should be closed now.

**Pull request closed without merging**